main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -36,6 +36,25 @@ import scala.collection.mutable
   */
 object Weeder {
 
+  /**
+    * Words that the Flix compiler reserves for special expressions.
+    * Users must not define fields or variables with these names.
+    */
+  private val ReservedWords = Set(
```

Comment:
Should this be in two?

Keywords versus types or?

Reply:
It could be, but it would add another bit of complexity. What's the rationale?

Reply:
Its fine. Lets keep it.